src/consumer/assigners/roundRobinAssigner/index.js

```diff
@@ -59,18 +63,16 @@ module.exports = ({ cluster }) => ({
       memberAssignment: MemberAssignment.encode({
```

Comment:
Shouldn't it provide the `userData` for that member in the `MemberAssignment`? The only difference is that the `assign` method gets the `userData` from each member (as `memberMetadata`). Maybe I'm missing something.

Reply:
The user data is part of the `members`, encoded into the `memberMetadata` by the `protocol` function.

The protocol function decides about the user data (in the RR assigner: nothing), and sends that to the broker as part of the member metadata. The broker then collects the members (and their metadata), and triggers the `assign` call (for the selected leader). The `assign` function implementation then call pull out the user data from the provided content.

The flow is indeed a bit odd, and I was arguing with myself here whether to leave some references lying around, but ultimately the RR assigner doesn't use any user data, and having it here is just confusing. Instead I added the unit test to show how to pass user data. Maybe a "how to write your own assigner" article could also help.